### DISCUSSION TITLE
How to batch actions?

### DISCUSSION THREAD

**Discussion by FigBug** [community/community#26205] [Actions]
<p>Is there any way to batch Actions? I’m migrating from Azure Pipelines which has the following option:</p>
<p><code>batch: boolean # batch changes if true (the default); start a new build for every push if false</code></p>
<p>Since mac builds are fairly expensive, I don’t want to run a build for each push. One at a time is enough. Or can I cancel all previous builds when a new push comes in?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/figbug">@figbug</a> ,</p>
<p>Sorry that ‘batch’ function same as azure pipeline currently is not supported in Github Actions. </p>
<p>And cannot cancel a previous workflow run automatically when a new push comes in. Have to manually cancel in the actions tab.</p>
<p>It’s similar with the <a href="https://github.community/t/quiet-period-for-builds/16914">ticket</a> which is answered.</p>
<p>Thanks.</p>

**Comment by n1hility** [comment]
<p>Hi <a class="mention" href="/figbug">@figbug</a> ,</p>
<p>I wrote an action to adress this problem, in case that it is useful to you:</p>
<p><a href="https://github.com/n1hility/cancel-previous-runs" rel="nofollow noopener">https://github.com/n1hility/cancel-previous-runs</a></p>

**Comment by FigBug** [comment]
<p>I ended up using this one: <a href="https://github.com/rokroskar/workflow-run-cleanup-action" rel="nofollow noopener">https://github.com/rokroskar/workflow-run-cleanup-action</a></p>
<p>I’ll check your out too to see how they compare.</p>
